src/App.jsx

```diff
@@ -1,33 +1,112 @@
+
+        console.log('trigger on apiUrl change');
+        const fetchData = async () => {
+            setLoading(true);
+
             try {
-                const response = await axios.get(baseUrl);
-                console.log(response.data);
-                setPokemonData(response.data);
+                //get the base url data
+                const listResponse = await axios.get(apiUrl);
+                //get the data object
+                const listData = listResponse.data;
 
-            } catch (e) {
-                console.error(e);
+                //get all de data for the 20 pokemons inside listData object and store in const detailResponses
+                //do a Promise.all to await all 20 calls to the api; the extra 20 calls could have been avoided if the api returned more data on the initial api call the first time around
+                const detailResponses = await Promise.all(
+                    listData.results.map(item => axios.get(item.url))
+                );
 
+                //iterate through detailResponses and store the data object into const detailedPokemon, this dataset is then used in the HTML rendering
+                const detailedPokemon = detailResponses.map(res => res.data);
+
+                //update states
+                setPokemonList(detailedPokemon);    //this state data is used in the rendering of the elements in the return() section
+                setNextUrl(listData.next);          //track the next url when user clicks on button 'Volgende'
+                setPrevUrl(listData.previous);      //track the next url when user clicks on button 'Vorige'
+            } catch (error) {
+                console.error("Something went wrong while loading Pokemon:", error);
+            } finally {
+                setLoading(false);
             }
+        };
+
+        fetchData();
+
+        return () => {
+            //abort http request when user cancels
+            controller.abort();
+        };
+
+    }, [apiUrl]);
+
+
+    const goBack = () => {
+        if (prevUrl) {
+            console.log('user clicked previous');
+            // Triggers a reload via apiUrl useEffect
+            setApiUrl(prevUrl);
         }
+    };
 
-        getPokemon();
-    }, []);
 
+    const goForward = () => {
+        if (nextUrl) {
+            console.log('user clicked next');
+            // Triggers a reload via apiUrl useEffect
+            setApiUrl(nextUrl);
+        }
+    };
 
+    //the HTML rendering
     return (
-        <>
-            <h1>Gotta catch em all!</h1>
-        </>
-    )
+        <section>
+            {/*show a 'Loading...' message*/}
+            {loading && <div className="spinner">Loading...</div>}
+            <div>
+                {/* navigation buttons using a component*/}
+                <div className="nav-buttons">
+                    <PaginationButtons onPrev={goBack} onNext={goForward} disablePrev={!prevUrl} disableNext={!nextUrl} />
```

Comment:
Het is beter om componenten te maken van enkelvoudige elementen. Dus geen paginationbutton**s** component, maar **twee losse Button-componenten** (zo blijft alles beter herbruikbaar!)